go/vt/vtctld/api.go

```diff
@@ -490,39 +466,14 @@ func initAPI(ctx context.Context, ts *topo.Server, actions *ActionRepository, he
 			if err := r.ParseForm(); err != nil {
 				return nil, err
 			}
-			keyspace := r.FormValue("keyspace")
-			cell := r.FormValue("cell")
 			tabletType := r.FormValue("type")
 			_, err := topoproto.ParseTabletType(tabletType)
 			// Excluding the case where parse fails because all tabletTypes was chosen.
 			if err != nil && tabletType != "all" {
 				return nil, fmt.Errorf("invalid tablet type: %v ", err)
 			}
-			metric := r.FormValue("metric")
 
-			// Setting default values if none was specified in the query params.
-			if keyspace == "" {
-				keyspace = "all"
-			}
-			if cell == "" {
-				cell = "all"
-			}
-			if tabletType == "" {
-				tabletType = "all"
-			}
-			if metric == "" {
-				metric = "health"
-			}
-
-			if healthcheck == nil {
-				return nil, fmt.Errorf("healthcheck not initialized")
-			}
-
-			heatmap, err := heatmapData(healthcheck, keyspace, cell, tabletType, metric)
-			if err != nil {
-				return nil, fmt.Errorf("couldn't get heatmap data: %v", err)
-			}
-			return heatmap, nil
+			return nil, fmt.Errorf("healthcheck not initialized")
```

Comment:
🤔 This change doesn't seem quite correct to me. Does this mean that the `tablet_statuses` endpoint will now always fail with a `healthcheck not initialized` failure?

Reply:
almost certainly yes; but, technically, it can also return with a "invalid tablet type" failure, which is a slightly different signal to the end user

Reply:
that said, that really only happens due to user input, so i'd be totally fine with the argument "yeah, but a user shouldn't do that so there's functionally no difference if we delete the endpoint" and immediately 404-ing